src/Files.App/Views/Pages/Properties.xaml.cs

```diff
@@ -1,269 +1,300 @@
+				Use your ResourceManager instance to create a ResourceContext as below. If you already have a ResourceManager instance,
+				replace the new instance created below with correct instance.
+				Read: https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/guides/mrtcore
+			*/new Microsoft.Windows.ApplicationModel.Resources.ResourceManager().CreateResourceContext().QualifierValues["LayoutDirection"];
+
+			if (flowDirectionSetting == "RTL")
+				FlowDirection = FlowDirection.RightToLeft;
+
+			contentFrame.Navigated += ContentFrame_Navigated;
+		}
+
+		protected override void OnNavigatedTo(NavigationEventArgs e)
+		{
+			var args = e.Parameter as PropertiesPageNavigationArguments;
+			AppInstance = args.AppInstanceArgument;
+			navParameterItem = args.Item;
+			listedItem = args.Item as ListedItem;
+			TabShorcut.Visibility = listedItem != null && listedItem.IsShortcutItem ? Visibility.Visible : Visibility.Collapsed;
+			TabLibrary.Visibility = listedItem != null && listedItem.IsLibraryItem ? Visibility.Visible : Visibility.Collapsed;
+			TabDetails.Visibility = listedItem != null && listedItem.FileExtension != null && !listedItem.IsShortcutItem && !listedItem.IsLibraryItem ? Visibility.Visible : Visibility.Collapsed;
+			TabSecurity.Visibility = args.Item is DriveItem ||
+				(listedItem != null && !listedItem.IsLibraryItem && !listedItem.IsRecycleBinItem) ? Visibility.Visible : Visibility.Collapsed;
+			TabCustomization.Visibility = listedItem != null && !listedItem.IsLibraryItem && (
+				(listedItem.PrimaryItemAttribute == Windows.Storage.StorageItemTypes.Folder && !listedItem.IsZipItem) ||
+				(listedItem.IsShortcutItem && !listedItem.IsLinkItem)) ? Visibility.Visible : Visibility.Collapsed;
+			TabCompatibility.Visibility = listedItem != null && (
+					".exe".Equals(listedItem is ShortcutItem sht ? System.IO.Path.GetExtension(sht.TargetPath) : listedItem.FileExtension, StringComparison.OrdinalIgnoreCase)
+				) ? Visibility.Visible : Visibility.Collapsed;
+			base.OnNavigatedTo(e);
+		}
+
+		private async void Properties_Loaded(object sender, RoutedEventArgs e)
+		{
+			AppSettings.ThemeModeChanged += AppSettings_ThemeModeChanged;
+			if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 8))
+			{
+				NavigationView.SizeChanged += NavigationView_SizeChanged;
+				appWindow.Destroying += AppWindow_Destroying;
+				await App.Window.DispatcherQueue.EnqueueAsync(() => AppSettings.UpdateThemeElements.Execute(null));
+			}
+			else
+			{
+				propertiesDialog = DependencyObjectHelpers.FindParent<ContentDialog>(this);
+				propertiesDialog.Closed += PropertiesDialog_Closed;
+			}
+		}
+
+		private void NavigationView_SizeChanged(object? sender, SizeChangedEventArgs? e)
+		{
+			int navigationViewWidth = 0;
+			bool overflowAdded = false;
+			foreach (NavigationViewItem item in NavigationView.MenuItems)
```

Comment:
This seems like an expensive calculation, it might be better to collapse the items that are hidden so that they don't take any space.

Reply:
The problem is that `NavigationView`, by default, takes more space than it's needed.
For example when there are only the first two tabs opened (and all other tabs are collapsed) NavView width is 320 while the two tabs total width is 209.
I was thinking of calculating `ActualWidth` once when the page is loaded and then when the selected tab changes, but if a page is hidden and the page get resized, DragRegion would be off

Reply:
Can you see if setting x:Load to false helps? For some reason hidden items are still taking up space but x:Load should solve that.

Reply:
Just learnt `x:Load` can't be used on `NavigationViewItem` (it throws a `XAML Parsing` exception).

Reply:
Can you add some comments to the code explaining the work arounds?